doc/deployment/deploy_custom.md

```diff
@@ -11,11 +22,360 @@ This document  details the various options of the `pachctl deploy custom ...` co
+because all the cloud providers use the standard etcd storage class.
+The VolumeClaimTemplate manifests is identical regardless of whether `aws`, `google` or `azure` is specified,
+for a StatefulSet.
+This is different for static volumes.
+
+### Object store configuration
+
+You can use the `--object-store` flag to configure Pachyderm to use one of two object-store drivers.
+The flag takes one argument, [which must be the `s3` value](https://github.com/pachyderm/pachyderm/issues/3996).
+This configuration uses the Amazon S3 driver to access your on-premises object store, 
+regardless of the vendor,
+since the Amazon S3 API is the standard with which every object store is designed to work.
+
+However, the S3 API has two different extant versions of *signature styles*, 
+which are how the object store validates client requests.
+S3v4 is the most current version,
+but many S3v2 object-store servers are still in use.
+Support for S3v2 is being deprecated. Therefore Pachyderm recommends using S3v4 in all deployments.
+
+If you need to access an object store that uses S3v2 signatures,
+you can specify the `--isS3V2` flag.
+
+This parameter configures Pachyderm to use the MinIO driver,
+allowing the use of the older signature.
+Also, this flag disables SSL for connections to the object store with the `minio` driver.
+You can reenable it with the `-s` or `--secure` flag.
+You can also edit the `pachyderm-storage-secret` Kubernetes manifest,
+as described below.
+
+#### Object store parameters
+
+The `--object-store` flag takes four (4) required, additional configuration arguments.
+These arguments must be placed immediately after [the persistent disk configuration parameters](#persistent-disk-parameters).
+
+- _bucket-name_: The name of the bucket, without the `s3://` prefix or a trailing forward slash (`/`).
+- _access-key_: The user access ID that is used to access the object store.
+- _secret-key_: The associated password that is used with the user access id to access the object store.
+- _endpoint_: The hostname and port that are used to access the object store, in <hostname>:<port> format.
+
+#### Example invocation with pv and object store
+The example on-premises cluster will use an on-premises MinIO object store with the following configuration parameters:
+  - SSL turned on,
+  - S3v4 signatures,
+  - the endpoint `minio:9000`
+  - the access key `OBSIJRBE0PP2NO4QOA27`,
+  - the secret key `tfteSlswRu7BJ86wekitnifILbZam1KYY3TG`,
+  - and a bucket named `pachyderm-bucket`.
+The deployment command uses the following flags:
+```
+pachctl deploy custom --persistent-disk aws --object-store s3 \
+    foobar 10 \
+    pachyderm-bucket  'OBSIJRBE0PP2NO4QOA27' 'tfteSlswRu7BJ86wekitnifILbZam1KYY3TG' 'minio:9000' \
+    --dynamic-etcd-nodes 10
+    [optional flags]
+```
+In the example command above, some of the arguments may contain characters that the shell could interpret in single-quotes.
+
+
+### Additional flags
```

Comment:
all the headings under this *Additional flags* can be put in a table or a list. We also need an intro text before the table. Something like:
```
This table lists the <blah> flags that you can use to customize your Pachyderm deployment.
```